metrics_v2/publisher/event_publisher.py

```diff
@@ -0,0 +1,200 @@
+  json_format.Parse(mcc_json, metric_config)
+
+  stackdriver_query = textwrap.dedent(f"""\
+    resource.type=k8s_container
+    resource.labels.project_id={project}
+    resource.labels.cluster_name=${cluster_name}
+    resource.labels.namespace_name={job.metadata.namespace}
+    resource.labels.pod_name:{job.metadata.name}
+    resource.labels.location:{cluster_location}
+  """)
+  stackdriver_link = "https://console.cloud.google.com/logs?{}".format(
+      urllib.parse.urlencode(
+          {'project': project, 'advancedFilter': stackdriver_query}))
+  
+  start_time = timestamp_pb2.Timestamp()
+  start_time.FromDatetime(job.status.start_time)
+  duration = duration_pb2.Duration()
+  duration.FromTimedelta(condition.last_transition_time - job.status.start_time)
+
+  return metrics_pb2.TestCompletedEvent(
+    benchmark_id=job.metadata.labels['benchmarkId'],
+    output_path=output_path,
+    status=job_status,
+    num_attempts=(job.status.succeeded or 0) + (job.status.failed or 0),
+    start_time=start_time,
+    duration=duration,
+    metric_collection_config=metric_config,
+    labels=job.metadata.labels,
+    debug_info=metrics_pb2.DebugInfo(
+      logs_link=stackdriver_link,
+      # TODO: fix hard-coded region and cluster name
+      details_link=f'https://console.cloud.google.com/kubernetes/job/{cluster_location}/{cluster_name}/{job.metadata.namespace}/{job.metadata.name}?project={project}'
+    )
+  )
+
+def main(argv):
+  try:
+    kubernetes.config.load_incluster_config()
+  except:
+    logging.warning("No cluster config found. Trying local kube config.")
+    kubernetes.config.load_kube_config()
+
+  resource_version = FLAGS.resource_version or _load_resource_version()
+
+  project = FLAGS.project or google.auth.default()[1]
+  cluster_name = FLAGS.cluster_name or _get_metadata('cluster-name')
+  cluster_location = FLAGS.cluster_location or _get_metadata('cluster-location')
+
+  topic = f'projects/{project}/topics/{FLAGS.pubsub_topic}'
+  publisher = pubsub_v1.PublisherClient()
+
+  while True:
+    try:
+      k8s_client = kubernetes.client.BatchV1Api()
+      job_watcher = kubernetes.watch.Watch()
+      event_stream = job_watcher.stream(
+          k8s_client.list_namespaced_job,
+          FLAGS.namespace,
+          label_selector='benchmarkId',
+          resource_version=resource_version)
+      for event in event_stream:
+        job = event['object']
+        if event['type'] != 'MODIFIED':
+          logging.info('Skipping event %s for Job %s', event['type'], job.metadata.name)
+          continue
+        elif job.status.active or not job.status.conditions:
+          logging.info('Job %s still active or has no conditions: %s', job.metadata.name, str(job.status))
+          continue
+
+        message = create_test_completed_event(job, FLAGS.model_output_bucket, cluster_name, cluster_location, project)
+        logging.info('No message to publish for %s.', job.metadata.name)
```

Comment:
should this line be e.g. in an `else` statement below the `if message` block?

Reply:
Yes. Done.